### PR TITLE
Make tokens in GIDGoogleUser KVO-compliant

### PR DESCRIPTION
**Pull request by Alex-4-Git**
To be consistent with GIDAuthentication, the token properties in GIDGoogleUser should be KVO-compliant.

### CODE REVIEW

File: GoogleSignIn/Sources/GIDGoogleUser.m [hunk 1]
```diff
@@ -102,36 +102,14 @@ - (GIDConfiguration *)configuration {
...
 }
 
 - (GIDToken *)refreshToken {
-  @synchronized(self) {
-    if (!_cachedRefreshToken) {
-      _cachedRefreshToken = [[GIDToken alloc] initWithTokenString:_authState.refreshToken
-                                                   expirationDate:nil];
-    }
-  }
   return _cachedRefreshToken;
```
[the remark this answers is not in the document]
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/GIDGoogleUser.m:109]: Done.

File: GoogleSignIn/Sources/GIDGoogleUser.m [hunk 2]
```diff
@@ -153,10 +131,25 @@ - (void)updateAuthState:(OIDAuthState *)authState
     _authentication = [[GIDAuthentication alloc] initWithAuthState:authState];
     _profile = profileData;
     
-    // These three tokens will be generated in the getter and cached .
-    _cachedAccessToken = nil;
-    _cachedRefreshToken = nil;
-    _cachedIdToken = nil;
+    [self sendKVONotificationsBeforeChanges];
+    [self updateTokensWithAuthState:authState];
```
[the remark this answers is not in the document]
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/GIDGoogleUser.m:135]: Done.

File: GoogleSignIn/Sources/GIDGoogleUser.m [hunk 3]
```diff
@@ -153,10 +124,26 @@ - (void)updateAuthState:(OIDAuthState *)authState
...
-    // These three tokens will be generated in the getter and cached .
-    _cachedAccessToken = nil;
-    _cachedRefreshToken = nil;
-    _cachedIdToken = nil;
+    [self updateTokensWithAuthState:authState];
+  }
+}
+
+- (void)updateTokensWithAuthState:(OIDAuthState *)authState {
+  self.accessToken = [[GIDToken alloc] initWithTokenString:authState.lastTokenResponse.accessToken
```
Comment (petea) [inline GoogleSignIn/Sources/GIDGoogleUser.m:132]: Before updating the value of these three properties, we should check that we're not about to set it to a new `GIDToken` instance that `isEqual` to the current value of the property.  The same would go for `nil` values for `idToken`.

File: GoogleSignIn/Sources/GIDGoogleUser.m [hunk 4]
```diff
@@ -42,12 +42,19 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+@interface GIDGoogleUser()
```
Comment (petea) [inline GoogleSignIn/Sources/GIDGoogleUser.m:45]: Add a space before `()`.

**Review by petea — CHANGES_REQUESTED** [review]

File: GoogleSignIn/Sources/GIDGoogleUser.m (hunk 3 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/GIDGoogleUser.m]: Done.

File: GoogleSignIn/Sources/GIDGoogleUser.m (hunk 4 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/GIDGoogleUser.m]: Done.

File: GoogleSignIn/Tests/Unit/GIDGoogleUserTest.m [hunk 5]
```diff
@@ -125,11 +124,23 @@ - (void)testUpdateAuthState {
   XCTAssertEqualObjects(user.idToken.tokenString, updatedIDToken);
   XCTAssertEqualWithAccuracy([user.idToken.expirationDate timeIntervalSince1970],
                              updatedIDTokenExpireTime, kTimeAccuracy);
+  XCTAssertEqualObjects(user.refreshToken.tokenString, kNewRefreshToken);
```
Comment (petea) [inline GoogleSignIn/Tests/Unit/GIDGoogleUserTest.m:127]: We should test that GIDToken instances are unchanged in the scenario in which the token strings are unchanged.

**Review by petea — CHANGES_REQUESTED** [review]

File: GoogleSignIn/Tests/Unit/GIDGoogleUserTest.m (hunk 5 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Tests/Unit/GIDGoogleUserTest.m:127]: Added a new test case `testUpdateAuthStateWithUnchangedRefreshTokenAndIDToken`

File: GoogleSignIn/Tests/Unit/GIDGoogleUserTest.m [hunk 6]
```diff
@@ -125,11 +124,48 @@ - (void)testUpdateAuthState {
...
   XCTAssertEqual(user.profile, updatedProfileData);
 }
 
+// When updating with a new OIDAuthState in which token information is not changed, the token object
```
Comment (petea) [inline GoogleSignIn/Tests/Unit/GIDGoogleUserTest.m:131]: object -> objects

**Review by petea — APPROVED** [review]

**Pull request merged**